### ISSUE TITLE
Release v1.1.3: OSDs failing to start in integration tests

### ISSUE THREAD

**Issue by travisn** [rook/rook#4164]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
This is blocking the v1.1.3 release.

In the release branch, we are seeing a consistent integration test failure. The OSD in the test is failing to start and causing the following failures. Other failures will certainly be reported as well, but the tests haven't completed yet. See this [test log](https://jenkins.rook.io/blue/rest/organizations/jenkins/pipelines/rook/pipelines/rook/branches/release-1.1/runs/80/nodes/48/steps/109/log/?start=0)
```
--- FAIL: TestCephBlockSuite (592.75s)
    --- FAIL: TestCephBlockSuite/TestBlockStorageMountUnMountForDifferentAccessModes (200.34s)
        require.go:1159: 
            	Error Trace:	ceph_block_test.go:173
            	            				ceph_block_test.go:229
            	Error:      	Should be true
            	Test:       	TestCephBlockSuite/TestBlockStorageMountUnMountForDifferentAccessModes
            	Messages:   	Make sure PVC is Bound
    --- FAIL: TestCephBlockSuite/TestBlockStorageMountUnMountForStatefulSets (210.64s)
        require.go:1159: 
            	Error Trace:	ceph_block_test.go:124
            	Error:      	Should be true
            	Test:       	TestCephBlockSuite/TestBlockStorageMountUnMountForStatefulSets
```

Here is the osd pod status and log showing the failure.
```
~/src/go/src/github.com/rook/rook$ kn get pod
NAME                                                           READY   STATUS      RESTARTS   AGE
rook-ceph-mgr-a-6fdc884cd-9bnqf                                1/1     Running     0          3m43s
rook-ceph-mon-a-5b964b995b-tdqxl                               1/1     Running     0          3m54s
rook-ceph-osd-0-7f7796d8b6-f6zzp                               0/1     Error       5          3m15s
rook-ceph-osd-prepare-6499238cece7ed09269e78f6b753bc7c-mxlxj   0/1     Completed   0          3m19s
rook-ceph-rbd-mirror-a-7694dcb8b9-ctmv4                        1/1     Running     0          3m14s
rook-ceph-tools                                                1/1     Running     0          3m10s
~/src/go/src/github.com/rook/rook$ klog osd-0
did not load config file, using default settings.
2019-10-23 21:12:25.463 7f04fd793dc0 -1 Errors while parsing config file!
2019-10-23 21:12:25.463 7f04fd793dc0 -1 parse_file: cannot open /etc/ceph/block-test-ns.conf: (2) No such file or directory
2019-10-23 21:12:25.463 7f04fd793dc0 -1 parse_file: cannot open /root/.ceph/block-test-ns.conf: (2) No such file or directory
2019-10-23 21:12:25.463 7f04fd793dc0 -1 parse_file: cannot open block-test-ns.conf: (2) No such file or directory
2019-10-23 21:12:25.463 7f04fd793dc0 -1 Errors while parsing config file!
2019-10-23 21:12:25.463 7f04fd793dc0 -1 parse_file: cannot open /etc/ceph/block-test-ns.conf: (2) No such file or directory
2019-10-23 21:12:25.463 7f04fd793dc0 -1 parse_file: cannot open /root/.ceph/block-test-ns.conf: (2) No such file or directory
2019-10-23 21:12:25.463 7f04fd793dc0 -1 parse_file: cannot open block-test-ns.conf: (2) No such file or directory
2019-10-23 21:12:25.485 7f04fd793dc0 -1 Falling back to public interface
2019-10-23 21:12:25.496 7f04fd793dc0 -1 filestore(/data/rook-test/block-test-ns/test-5577006791947779410/osd0) WARNING: max attr value size (1024) is smaller than osd_max_object_name_len (2048).  Your backend filesystem appears to not support attrs large enough to handle the configured max rados name size.  You may get unexpected ENAMETOOLONG errors on rados operations or buggy behavior
2019-10-23 21:12:25.496 7f04fd793dc0 -1 filestore(/data/rook-test/block-test-ns/test-5577006791947779410/osd0) WARNING: max attr value size (1024) is smaller than osd_max_object_name_len (2048).  Your backend filesystem appears to not support attrs large enough to handle the configured max rados name size.  You may get unexpected ENAMETOOLONG errors on rados operations or buggy behavior
2019-10-23 21:12:25.520 7f04fd793dc0 -1 journal FileJournal::_open: disabling aio for non-block journal.  Use journal_force_aio to force use of aio anyway
2019-10-23 21:12:25.521 7f04fd793dc0 -1 journal do_read_entry(8192): bad header magic
2019-10-23 21:12:25.521 7f04fd793dc0 -1 journal do_read_entry(8192): bad header magic
2019-10-23 21:12:25.521 7f04fd793dc0 -1 osd.0 0 backend (filestore) is unable to support max object name[space] len
2019-10-23 21:12:25.521 7f04fd793dc0 -1 osd.0 0    osd max object name len = 2048
2019-10-23 21:12:25.521 7f04fd793dc0 -1 osd.0 0    osd max object namespace len = 256
2019-10-23 21:12:25.521 7f04fd793dc0 -1 osd.0 0 (36) File name too long
2019-10-23 21:12:25.533 7f04fd793dc0 -1  ** ERROR: osd init failed: (36) File name too long
```

Reverting the change to remove the `--conf` flag (#4161) is allowing the osd to start successfully and pass the integration tests locally.

**Expected behavior:**
The OSDs should start successfully and allow the integration tests to pass.

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):

**Comment by BlaineEXE** [comment]
I would guess that the dir-based OSDs *do* need the `--conf` flag when the changes from #3554 are not also present.

**Comment by travisn** [comment]
Instead of removing the `--conf` flag, in the release branch it would appear we need the changes as started in #4129. Those changes weren't deemed necessary in master since #3554 had been merged, but they are necessary in the release branch without #3554.

**Comment by travisn** [comment]
Fixed by #4165, though the v1.1.3 release was skipped because of this issue
